### PR TITLE
capdebuginfo: Added new script

### PR DESCRIPTION
**Pull request by arshadaleem66**
JIRA : SB-10632

Capdebuginfo script is added at meta-mentor/scripts/ The script
collects host and build information and zips into a tar file
which the customer can attach with support ticket for
debugging purpose at our side.

Signed-off-by: arshadaleem66 <arshad_aleem@mentor.com>

### CODE REVIEW

**Comment by arshadaleem66** [comment]
@Noor-Ahsan  Please don't Merge this Pull Request. I will update it with changes in the script. I will add more commits to this PR.

**Comment by kergoth** [comment]
Just a note for future reference, there's no need to create a new pull request to make changes, you can just push to the existing branch, and the existing pull request will update accordingly.

**Comment by arshadaleem66** [comment]
@kergoth, @ahsanhussain @Noor-Ahsan 

Please have a look at the script now.

File: scripts/capdebuginfo [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+    if [ -n "$a" ]
+    then
+      break
+    fi
+done
+
+MELDIR="${a%%/bitbake/bin}"
+
+
+
```
Comment (ahsanhussain) [inline scripts/capdebuginfo]: Remove extra lines here.

File: scripts/capdebuginfo [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+   exit 1
+else
+   # Remove output directory after creating tarball archive file
+   if [ -d "${OUTPUTDIRNAME}" ]; then
+      rm -rf ${OUTPUTDIRNAME}
+   fi
+   echo "Done!!!"
+   echo -en "\nPlease attach ${OUTPUTDIRNAME}.tar.bz2 to your Support Request ticket.\n\n"
+fi
+fi
```
Comment (ahsanhussain) [inline scripts/capdebuginfo]: Please fix indentation for the outer if condition.

File: scripts/capdebuginfo [hunk 3]
```diff
@@ -0,0 +1,294 @@
...
+END
+}
+
+
+abspath () {
+    readlink -e "${1}"
+}
+setup_environment() {
+if("Ubuntu"=`lsb_release -i -s`) then
+   source "${MELDIR}"/meta-mentor/scripts/setup-ubuntu
```
Comment (ahsanhussain) [inline scripts/capdebuginfo]: Add a check here if _for some reason_ sourcing fails; I tried this on dogwood, it was unable to source due to path difference, so I got dpkg-query results for all the packages on host.

File: scripts/capdebuginfo [hunk 4]
```diff
@@ -0,0 +1,294 @@
...
+   local __outputdir="${1}"
+   local __output="${1}/host/${2}"
+   local __hostos_distid=`lsb_release -i -s`
+   local __hostos_distvers=`lsb_release -r -s`
+   local __pkgs
+   
+   echo -en "\nHost OS is ${__hostos_distid} ${__hostos_distvers}:\n" >> ${__output}
+   case "${__hostos_distid}" in
+      Ubuntu)
+	 source "${MELDIR}"/meta-mentor/scripts/setup-ubuntu
```
Comment (ahsanhussain) [inline scripts/capdebuginfo]: Isn't it already sourced above in setup_environment()?

File: scripts/capdebuginfo [hunk 5]
```diff
@@ -0,0 +1,294 @@
...
+   mkdir ${__outputdir}/mel/${__arch}
+   cp -a ${__pkgdir} ${__outputdir}/mel/${__arch}/.
+}
+
+
+########################################################################################################
+#
+# Start execution of script
+#
+#set -x
```
Comment (ahsanhussain) [inline scripts/capdebuginfo]: is this left intentionally or debugging residue?

**Comment by kergoth** [comment]
Output of shellcheck, for reference:
```
In scripts/capdebuginfo line 19:
WORKDIR=${PWD}
^-- SC2034: WORKDIR appears unused. Verify it or export it.


In scripts/capdebuginfo line 23:
SCRIPT_NAME="$(basename ${BASH_SOURCE})"
                        ^-- SC2128: Expanding an array without an index only gives the first element.
                        ^-- SC2086: Double quote to prevent globbing and word splitting.


In scripts/capdebuginfo line 54:
if("Ubuntu"=`lsb_release -i -s`) then
            ^-- SC2046: Quote this to prevent word splitting.
            ^-- SC2006: Use $(..) instead of legacy `..`.


In scripts/capdebuginfo line 55:
   source "${MELDIR}"/meta-mentor/scripts/setup-ubuntu
   ^-- SC1090: Can't follow non-constant source. Use a directive to specify location.


In scripts/capdebuginfo line 57:
   source "${MELDIR}"/meta-mentor/scripts/setup-rh
   ^-- SC1090: Can't follow non-constant source. Use a directive to specify location.


In scripts/capdebuginfo line 69:
               echo "Invalid input for option -"${opt}""
                                                ^-- SC2027: The surrounding quotes actually unquote this. Remove or escape them.
                                                ^-- SC2086: Double quote to prevent globbing and word splitting.


In scripts/capdebuginfo line 73:
               BUILDERR=$(find "${BUILDDIR}/tmp/work/" -maxdepth 2 -type d -name "${OPTARG}")
                                ^-- SC2153: Possible misspelling: BUILDDIR may not be assigned, but BUILDERR is.


In scripts/capdebuginfo line 77:
            echo "Option -"${OPTARG}" requires an argument." >&2
                           ^-- SC2027: The surrounding quotes actually unquote this. Remove or escape them.
                           ^-- SC2086: Double quote to prevent globbing and word splitting.


In scripts/capdebuginfo line 80:
            echo "Invalid option: -"${OPTARG}"" >&2
                                    ^-- SC2027: The surrounding quotes actually unquote this. Remove or escape them.
                                    ^-- SC2086: Double quote to prevent globbing and word splitting.


In scripts/capdebuginfo line 97:
   echo -en "\nHost Distro Info:\n" >> ${__output}
             ^-- SC1117: Backslash is literal in "\n". Prefer explicit escaping: "\\n".
                                ^-- SC1117: Backslash is literal in "\n". Prefer explicit escaping: "\\n".
                                       ^-- SC2086: Double quote to prevent globbing and word splitting.


In scripts/capdebuginfo line 98:
   lsb_release -a >> "${__output}" 2> /dev/null
   ^-- SC2129: Consider using { cmd1; cmd2; } >> file instead of individual redirects.


In scripts/capdebuginfo line 99:
   echo -en "\nHost System Info:\n" >> "${__output}"
             ^-- SC1117: Backslash is literal in "\n". Prefer explicit escaping: "\\n".
                                ^-- SC1117: Backslash is literal in "\n". Prefer explicit escaping: "\\n".


In scripts/capdebuginfo line 101:
   echo -en "\nHost env Info:\n" >> "${__output}"
             ^-- SC1117: Backslash is literal in "\n". Prefer explicit escaping: "\\n".
                             ^-- SC1117: Backslash is literal in "\n". Prefer explicit escaping: "\\n".


In scripts/capdebuginfo line 103:
   echo -en "\nHost disk space usage:\n" >> "${__output}"
             ^-- SC1117: Backslash is literal in "\n". Prefer explicit escaping: "\\n".
                                     ^-- SC1117: Backslash is literal in "\n". Prefer explicit escaping: "\\n".


In scripts/capdebuginfo line 105:
   echo -en "\nHost memory usage:\n" >> "${__output}"
             ^-- SC1117: Backslash is literal in "\n". Prefer explicit escaping: "\\n".
                                 ^-- SC1117: Backslash is literal in "\n". Prefer explicit escaping: "\\n".


In scripts/capdebuginfo line 107:
   echo -en "\n" >> "${__output}"
             ^-- SC1117: Backslash is literal in "\n". Prefer explicit escaping: "\\n".


In scripts/capdebuginfo line 119:
   local __license=`printenv "${3}"`
         ^-- SC2155: Declare and assign separately to avoid masking return values.
                   ^-- SC2006: Use $(..) instead of legacy `..`.


In scripts/capdebuginfo line 121:
   echo -en "\n${__licensename}:\n" >> ${__output}
             ^-- SC1117: Backslash is literal in "\n". Prefer explicit escaping: "\\n".
                                ^-- SC1117: Backslash is literal in "\n". Prefer explicit escaping: "\\n".
                                       ^-- SC2086: Double quote to prevent globbing and word splitting.


In scripts/capdebuginfo line 124:
      echo -en "${__license}\n" >> "${__output}"
                            ^-- SC1117: Backslash is literal in "\n". Prefer explicit escaping: "\\n".


In scripts/capdebuginfo line 126:
      for i in $(echo "${__license//:/$'\n'}"); do
               ^-- SC2116: Useless echo? Instead of 'cmd $(echo foo)', just use 'cmd foo'.


In scripts/capdebuginfo line 133:
            echo -en "\n${i} is a license server\n" >> ${__output}
                      ^-- SC1117: Backslash is literal in "\n". Prefer explicit escaping: "\\n".
                                                ^-- SC1117: Backslash is literal in "\n". Prefer explicit escaping: "\\n".
                                                       ^-- SC2086: Double quote to prevent globbing and word splitting.


In scripts/capdebuginfo line 136:
            echo -en "\n${i} does not seem to be a valid entry\n" >> "${__output}"
                      ^-- SC1117: Backslash is literal in "\n". Prefer explicit escaping: "\\n".
                                                              ^-- SC1117: Backslash is literal in "\n". Prefer explicit escaping: "\\n".


In scripts/capdebuginfo line 140:
      echo -en "\n${__licensename} is not set!\n" >> "${__output}"
                ^-- SC1117: Backslash is literal in "\n". Prefer explicit escaping: "\\n".
                                              ^-- SC1117: Backslash is literal in "\n". Prefer explicit escaping: "\\n".


In scripts/capdebuginfo line 151:
   local __hostos_distid=`lsb_release -i -s`
         ^-- SC2155: Declare and assign separately to avoid masking return values.
                         ^-- SC2006: Use $(..) instead of legacy `..`.


In scripts/capdebuginfo line 152:
   local __hostos_distvers=`lsb_release -r -s`
         ^-- SC2155: Declare and assign separately to avoid masking return values.
                           ^-- SC2006: Use $(..) instead of legacy `..`.


In scripts/capdebuginfo line 155:
   echo -en "\nHost OS is ${__hostos_distid} ${__hostos_distvers}:\n" >> ${__output}
             ^-- SC1117: Backslash is literal in "\n". Prefer explicit escaping: "\\n".
                                                                  ^-- SC1117: Backslash is literal in "\n". Prefer explicit escaping: "\\n".
                                                                         ^-- SC2086: Double quote to prevent globbing and word splitting.


In scripts/capdebuginfo line 158:
	 source "${MELDIR}"/meta-mentor/scripts/setup-ubuntu
         ^-- SC1090: Can't follow non-constant source. Use a directive to specify location.


In scripts/capdebuginfo line 160:
         dpkg-query -W -f='${binary:Package} ${Version} ${db:Status-Status}\n' ${__pkgs} >> ${__output}
                                                                               ^-- SC2086: Double quote to prevent globbing and word splitting.
                                                                                            ^-- SC2086: Double quote to prevent globbing and word splitting.


In scripts/capdebuginfo line 163:
         source "${MELDIR}"/meta-mentor/scripts/setup-rh
         ^-- SC1090: Can't follow non-constant source. Use a directive to specify location.


In scripts/capdebuginfo line 164:
         __pkgs="$packages"
                 ^-- SC2154: packages is referenced but not assigned.


In scripts/capdebuginfo line 165:
         rpm -q ${__pkgs} >> ${__output}
                ^-- SC2086: Double quote to prevent globbing and word splitting.
                             ^-- SC2086: Double quote to prevent globbing and word splitting.


In scripts/capdebuginfo line 179:
      cp -a ${HOMEDIR}/.mentor/.registry ${__outputdir}/host/mentor-registry
            ^-- SC2086: Double quote to prevent globbing and word splitting.
                                         ^-- SC2086: Double quote to prevent globbing and word splitting.


In scripts/capdebuginfo line 181:
      echo -en "\nNo MEL install info found!\n" >> ${__output}
                ^-- SC1117: Backslash is literal in "\n". Prefer explicit escaping: "\\n".
                                            ^-- SC1117: Backslash is literal in "\n". Prefer explicit escaping: "\\n".
                                                   ^-- SC2086: Double quote to prevent globbing and word splitting.


In scripts/capdebuginfo line 195:
   local __arch="$(echo ${__builderr#/*/work/} | cut -d"/" -f1)"
         ^-- SC2155: Declare and assign separately to avoid masking return values.
                        ^-- SC2086: Double quote to prevent globbing and word splitting.


In scripts/capdebuginfo line 196:
   local __pkgname="$(echo ${__builderr#/*/work/} | cut -d"/" -f2)"
         ^-- SC2155: Declare and assign separately to avoid masking return values.
                           ^-- SC2086: Double quote to prevent globbing and word splitting.


In scripts/capdebuginfo line 199:
   cp -a ${__builddir}/buildhistory/images/ ${__outputdir}/mel/.
         ^-- SC2086: Double quote to prevent globbing and word splitting.
                                            ^-- SC2086: Double quote to prevent globbing and word splitting.


In scripts/capdebuginfo line 200:
   cp -a ${__builddir}/tmp/log/ ${__outputdir}/mel/.
         ^-- SC2086: Double quote to prevent globbing and word splitting.
                                ^-- SC2086: Double quote to prevent globbing and word splitting.


In scripts/capdebuginfo line 201:
   cp -a ${__builddir}/conf/ ${__outputdir}/mel/.
         ^-- SC2086: Double quote to prevent globbing and word splitting.
                             ^-- SC2086: Double quote to prevent globbing and word splitting.


In scripts/capdebuginfo line 203:
   mkdir ${__outputdir}/mel/${__arch}
         ^-- SC2086: Double quote to prevent globbing and word splitting.
                            ^-- SC2086: Double quote to prevent globbing and word splitting.


In scripts/capdebuginfo line 204:
   cp -a ${__pkgdir} ${__outputdir}/mel/${__arch}/.
         ^-- SC2086: Double quote to prevent globbing and word splitting.
                     ^-- SC2086: Double quote to prevent globbing and word splitting.
                                        ^-- SC2086: Double quote to prevent globbing and word splitting.


In scripts/capdebuginfo line 225:
      echo -en "\nERROR:  Input argument was invalid!!!\n\n"
                ^-- SC1117: Backslash is literal in "\n". Prefer explicit escaping: "\\n".
                                                       ^-- SC1117: Backslash is literal in "\n". Prefer explicit escaping: "\\n".
                                                         ^-- SC1117: Backslash is literal in "\n". Prefer explicit escaping: "\\n".


In scripts/capdebuginfo line 227:
elif [ -z "${@}" ]; then
          ^-- SC2198: Arrays don't work as operands in [ ]. Use a loop (or concatenate with * instead of @).


In scripts/capdebuginfo line 228:
   echo -en "\nWARNING:  No input argument was given on command line.\n\n"
             ^-- SC1117: Backslash is literal in "\n". Prefer explicit escaping: "\\n".
                                                                     ^-- SC1117: Backslash is literal in "\n". Prefer explicit escaping: "\\n".
                                                                       ^-- SC1117: Backslash is literal in "\n". Prefer explicit escaping: "\\n".


In scripts/capdebuginfo line 234:
   read IN1
   ^-- SC2162: read without -r will mangle backslashes.


In scripts/capdebuginfo line 235:
   BUILDERR="$(find ${BUILDDIR}/tmp/work/ -maxdepth 2 -type d -name "${IN1}")"
                    ^-- SC2086: Double quote to prevent globbing and word splitting.


In scripts/capdebuginfo line 238:
      echo -en "\nERROR (option -e):  Invalid package name!!!\n\n"
                ^-- SC1117: Backslash is literal in "\n". Prefer explicit escaping: "\\n".
                                                             ^-- SC1117: Backslash is literal in "\n". Prefer explicit escaping: "\\n".
                                                               ^-- SC1117: Backslash is literal in "\n". Prefer explicit escaping: "\\n".


In scripts/capdebuginfo line 242:
for i in $(echo $PATH | tr ":" "\n")
                ^-- SC2086: Double quote to prevent globbing and word splitting.
                                ^-- SC1117: Backslash is literal in "\n". Prefer explicit escaping: "\\n".


In scripts/capdebuginfo line 244:
    a="$(echo $i | grep "/bitbake/bin")"
              ^-- SC2086: Double quote to prevent globbing and word splitting.


In scripts/capdebuginfo line 273:
get_melbuildinfo ${OUTPUTDIRNAME} ${BUILDERR} ${BUILDDIR}
                                  ^-- SC2086: Double quote to prevent globbing and word splitting.
                                              ^-- SC2086: Double quote to prevent globbing and word splitting.


In scripts/capdebuginfo line 280:
if [[ ${?} -ne 0 ]]; then
      ^-- SC2181: Check exit code directly with e.g. 'if mycmd;', not indirectly with $?.


In scripts/capdebuginfo line 281:
   echo -en "\ntar errno: ${?}\n" >> ${OUTPUTDIRNAME}/host/${HOSTINFOFILE}
             ^-- SC1117: Backslash is literal in "\n". Prefer explicit escaping: "\\n".
                              ^-- SC1117: Backslash is literal in "\n". Prefer explicit escaping: "\\n".


In scripts/capdebuginfo line 292:
   echo -en "\nPlease attach ${OUTPUTDIRNAME}.tar.bz2 to your Support Request ticket.\n\n"
             ^-- SC1117: Backslash is literal in "\n". Prefer explicit escaping: "\\n".
                                                                                     ^-- SC1117: Backslash is literal in "\n". Prefer explicit escaping: "\\n".
                                                                                       ^-- SC1117: Backslash is literal in "\n". Prefer explicit escaping: "\\n".

```

**Pull request merged**